contracts/src/gateway/GatewayManagerFacet.sol

```diff
@@ -114,7 +118,11 @@ contract GatewayManagerFacet is GatewayActorModifiers, ReentrancyGuard {
 
         s.totalSubnets -= 1;
         delete s.subnets[id];
-        s.subnetKeys.remove(id);
+
+        bool removed = s.subnetKeys.remove(id);
+        if (!removed) {
+            revert NotInSet();
```

Comment:
What is the scenario when this happens?

Reply:
A catastrophic scenario is when the integrity of the actors is violated. So it is the same as panicking in Go in case we do not know what to do.

Reply:
I see your point. However, IMO we should be handling cases for which we see _any_ practical possibility.
For example: are you able, using the contract's API to produce a test case which leads to this scenario?

(hm... btw in web2 we could at least add some kind of "WTF" log in this case, but in smart contracts we're just reverting a user transaction and have no way to learn from this)

Reply:
We return the specific errors (I introduced) that are returned only in this case.

Reply:
But we have no way to monitor whether they were thrown without running a node on a subnet and having some error filters set up.

Reply:
Correct as for all other errors we return using revert.

Reply:
@dnkolegov can you take a look at this question above?
> For example: are you able, using the contract's API to produce a test case which leads to this scenario?

Reply:
I can't do that. If it were possible, we would have a bug in the code. I am fine with removing those checks.